src/os_common/os_alloc.c

```diff
@@ -129,9 +150,22 @@ __realloc_func(WT_SESSION_IMPL *session, size_t *bytes_allocated_ret, size_t byt
             WT_STAT_CONN_INCR(session, memory_grow);
     }
 
-    if ((p = realloc(p, bytes_to_allocate)) == NULL)
-        WT_RET_MSG(session, __wt_errno(), "memory allocation of %" WT_SIZET_FMT " bytes failed",
-          bytes_to_allocate);
+    /*
+     * If realloc_malloc is enabled, use malloc to mimic realloc functionality, copy to the new
+     * memory, scribble over the old memory then free it.
+     */
+    if (realloc_malloc) {
+        if ((p = malloc(bytes_to_allocate)) == NULL)
+            WT_RET_MSG(session, __wt_errno(), "memory allocation of %" WT_SIZET_FMT " bytes failed",
+              bytes_to_allocate);
+        memcpy(p, tmpp, *bytes_allocated_ret);
```

Comment:
Isn't `bytes_allocated` the same as `*bytes_allocated_ret`?  Should be clearer.

Reply:
`bytes_allocated` will only be the same as `*bytes_allocated_ret` if `bytes_allocated_ret` is not NULL. Since we've merged the realloc_malloc functionality into `__realloc_func`, in the normal realloc case `bytes_allocated_ret` could still be NULL.

Reply:
Right you are!